packages/react-redux/src/slices/cartoSlice.js

```diff
@@ -148,10 +149,15 @@ export const createCartoSlice = (initialState) => {
         }
       },
       removeFilter: (state, action) => {
-        const { id, column } = action.payload;
+        const { id, column, type } = action.payload;
         const source = state.dataSources[id];
 
         if (source && source.filters && source.filters[column]) {
+          if (type && source.filters[column][type]) {
+            delete source.filters[column][type];
```

Comment:
After this delete I would check if `source.filters[column]` has more filters and if not I would execute `delete source.filters[column]`

Reply:
Agreed.

Related, after having worked for some weeks close to this topic, I think the way we store filters could be improved. The 2-levels nested structure is complicated to work with. They are complex to iterate, and we always have to bother about whether the objects exists, or when to completely removing them when they are empty…
I would love to have a flat array here, with `[ ]` as default and easy filtering.

Reply:
done ✅